### DISCUSSION TITLE
增加节点后如何更新布局？

### DISCUSSION THREAD

**Discussion by lvchenyang** [antvis/X6#2537] [Q&A]
目前的需求是通过添加节点进行审批流程图的定义，采用的布局是`DagreLayout`，在第一次渲染的时候，动态计算了所有节点的layer，并进行了首次渲染，如下图所示：
<img width="920" alt="image" src="https://user-images.githubusercontent.com/5507872/185273864-6d6de3a5-b6c0-4603-8ca8-3221e0995f92.png">

现在的问题是：添加一个节点后，如何从新计算每个节点的layer? 或者如何操作让graph重新布局？目前添加节点后的视图如下所示：
<img width="1004" alt="image" src="https://user-images.githubusercontent.com/5507872/185274092-29f1c203-025e-469e-9f21-69259c7fffcf.png">

> 注：禁止了用户拖动节点自定义布局，所以每次增删节点后，都需要自动更新布局。

**Comment by zmcode** [comment]
将新增的节点id缓存, 将所有数据重新计算, 获取每个节点的x,y,  通过id拿到新增节点的x和y, 插入

**Reply by gufengbangye** [reply]
请问一下 我感觉算出来的坐标不对 是我需要转换一下么
